freyja/utils.py

```diff
@@ -39,6 +43,108 @@ def checkConfig(config):
+    df.columns = [dfc.split(' (')[0] for dfc in df.columns]
+    df = df.set_index('Date')
+
+    df = df.dropna(axis=0, how='all')
+    df = df.fillna(0)
+    df = df/100.
+    # df['Other'] = 1. - df['Delta']- df['Omicron']- df['Alpha']
+    # df2 = df[['Alpha','Delta','Omicron','Other']]
+    # df = df.drop(columns=['Alpha','Delta','Omicron','Other'])
+    # df.set_index('Date', inplace=True)
+    nBack = next((x[0]+1 for x in enumerate(df.index[::-1])
+                 if (df.index[-1] - x[1]).days > 56), 0)
+    rel_growth_rate = {
+        'Lineage': [],
+        'Estimated Advantage': [],
+        'Bootstrap 95% interval': [],
+    }
+    # get all lineages present at >0.1% average over last 8 weeks
+    lineages = df.columns[df.iloc[-nBack:].mean(axis=0) > 0.001]
+    for k, lineage in tqdm.tqdm(enumerate(lineages)):
+        days = np.array([(dfi - df.index[-nBack]).days
+                         for j, dfi in enumerate(df.index[-nBack:])])
+        data = df[lineage][-len(days):]
+        fit, covar = curve_fit(
+            f=logistic_growth,
+            xdata=days,
+            ydata=data,
+            p0=[0.003, 0.008],
+            bounds=([-1000] * 2, [1000] * 2)
+        )
+
+        # build 95% CI by bootstrapping.
+
+        bootSims = []
+        coef_ests = []
+        for j in range(nboots):
+            bootInds = np.random.randint(0, len(days), len(days))
+            daysBoot = days[bootInds]
+            dataBoot = data[bootInds]
+            try:
+                fitBoot, covarBoot = curve_fit(
+                    f=logistic_growth,
+                    xdata=daysBoot,
+                    ydata=dataBoot,
+                    p0=[100, 0.1],
+                    bounds=([-1000] * 2, [1000] * 2)
+                )
+            except RuntimeError:
+                print('WARNING: Optimal parameters not found: The maximum' +
+                      ' number of function evaluations is exceeded.')
+
+            bootSims.append([logistic_growth(i, *fitBoot) for i in days])
+            coef_ests.append(fitBoot[1])
+
+        bootSims = np.array(bootSims)
+        # boot_lower = np.percentile(bootSims,2.5,axis=0)
+        # boot_upper = np.percentile(bootSims,97.5,axis=0)
+
+        coef_lower = np.percentile(coef_ests, 2.5)
+        coef_upper = np.percentile(coef_ests, 97.5)
+
+        # fig, ax = plt.subplots()
+        # fit_ = [logistic_growth(i, *fit) for i in days]
+        # ax.plot(df.index[-len(days):],fit_,color=colors[k])
+        # ax.plot(df.index[-len(days):],data,'o',color=colors[k])
+        # ax.fill_between(df.index[-len(days):],boot_lower,boot_upper,alpha=0.4,color=colors[k])
+        # locator =mdates.MonthLocator(bymonthday=1)
+        # ax.xaxis.set_major_locator(locator)
+        # ax.xaxis.set_major_formatter(mdates.ConciseDateFormatter(locator))
+        # ax.set_ylabel('Prevalence')
+        # plt.savefig('fitting_'+lineage+'.pdf')
+
+        rate0 = fit[1]
+        serial_interval = 5.5
```

Comment:
looks like serial interval is still hardcoded here, so the user specified value doesn't get used

Reply:
Oops, I removed the line.